src/frontend/test_runner/tests/testdata/subquery_expr_correlated.yaml

```diff
@@ -151,3 +151,20 @@
                 LogicalScan { table: t1, columns: [_row_id#0, x, y] }
                 LogicalProject { exprs: [1:Int32, $0], expr_alias: [1,  ] }
                   LogicalScan { table: t2, columns: [y] }
+- sql: |
+    create table t1(x int, y int);
+    create table t2(x int, y int);
+    select x from t1 where y in (select y from t2 where t1.x = t2.x);
```

Comment:
Can you add more tests? e.g.

- `not in`
- with other conditions in the subquery
- with more than 1 correlated variable in the subquery
- with non-equi correlated condition in the subquery
- ...

Reply:
I have added tests for `not in` in `subquery_expr.yaml`, because we cann't unnest LogicalAntiApply now. I will add other types of tests.

Reply:
It's okay. You may add a test with error output if that's not supported yet.

Reply:
+1 for an error test

Reply:
I just tried it and it will panic, so it doesn't display any error. I just asked @skyzh about this, and he said that planner test can't handle panic.

Reply:
```rust
impl ToStream for LogicalApply {
    fn to_stream(&self) -> PlanRef {
        panic!("LogicalApply should be unnested")
    }

    fn logical_rewrite_for_stream(&self) -> (PlanRef, ColIndexMapping) {
        panic!("LogicalApply should be unnested")
    }
}
```

Will panic instead of returning an error. Planner test won't handle panic for now. If we want to test unsupported cases, we will need to refactor this into an error.

Reply:
I would recommend to returns an `Err` to user instead of `panic!` or `todo!`. 

As far as I know, no products can unnest all kinds of corrected subqueries, so we **have** to tell users "this query is not supported balabala"

Reply:
+1, I think we can do this in future PR.

Reply:
We have found that create_apply_or_join is ok to add a check. I will add an error report there.😇😇😇😇

Reply:
rethinking the issue, I consider we can do handle the unsupported sql in binder and planer and in optimizer we can panic. for the unnest case, I think `create_apply_or_join` is a good place in planner to return error

Reply:
`create_apply_or_join` may not be enough. This issue is mostly specific to `ExprImpl::Subquery`, `ExprImpl::CorrelatedInputRef`, and `LogicalApply`: they are supposed to be present in the expr tree or plan tree in early phases, and completely replaced later. Another way to report errors is to add a dedicate phase than scans the whole plan for any undesired residuals - if we do not want to make existing `to_prost` / `to_batch` to return `Result`.

Reply:
>  Another way to report errors is to add a dedicate phase than scans the whole plan for any undesired residuals

Do you mean do this check right after unnesting? If so, then we should add optimizer_error.

Reply:
I meant impl `PlanVisitor` just to check the existence of one of the three, and call it in `PlanRoot::gen_batch_query_plan` / `PlanRoot::gen_stream_plan` immediately after `gen_optimized_logical_plan`.

Reply:
In conclusion, we will add optimizer error or something else in future prs to report error instead of panicing.

Reply:
> impl PlanVisitor just to check the existence of one of the three
+1